adagrams/game.py

```diff
@@ -1,11 +1,149 @@
+        #build a data sturcture for the letter pool
+    #there are only 2 available C letters
+        #draw_letters cannot return more than 2 C's
+    #there are 12 E's but only one Z
+        #should be 12 times as likely to draw a E as a Z
+    #invoking this function should NOT change the pool of letters
+        #imagine user returns their hand to the pool before drawing new letterss
+    return drawn_letters
+
+#terminal vs code wave one success! -- terminal wave one success!
 
+#WAVE 2
 def uses_available_letters(word, letter_bank):
-    pass
+    #pass
+    #nOT_cAsE_sEnsiTiVe = word.upper() -- could have made a variable
+    #letter bank describes an array of drawn letters in a hand
+    for letter in list(word.upper()):
+    # set is a dict unordered duplicates not allowed -- set would work if using with list, tuple or dict
+    #set didn't work was not iterating thru for loop
+    # had to turn set into a list
+    #check if an input word only uses characters that are within a hand of drawn cards
+    #word describes some input word, and is a string
+        if word.upper().count(letter) > letter_bank.count(letter):
+            return False
+    #return False if not; if there is a letter in input that is not present in the letter_bank
+    #return True if every letter in the input word is available in the letter_bank
+    return True
 
+
+#WAVE 3
+#has one parameter word
 def score_word(word):
-    pass
+    # each letter within word has point value.
+    #number of points of each letter is summed up to rep the total score of word
+    # each letter's point value is described in VALUE_SCORES
+    score = sum(VALUE_SCORES.get(letter.upper(), 0) for letter in word)
+    #if the len of the word is 7,8,9,10 
+    if len(word) in [7, 8, 9, 10]:
+        # then the word gets an addtl 8 points
+        score += 8
+    #returns int rep number of points
+    return score
+
+#terminal vs code wave three success! -- terminal wave three success!
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+
+    max_score = 0
+    #find highest scoring word
+    max_word = ''
+    for word in word_list:
+        #for i in word:
+    # list of word list - one parameter - list of strings
+            word_score = score_word(word)
+        # calculates which of these words has the highest score, applies any ties
+            if word_score > max_score:
+                max_score = word_score
+                max_word = word
+            elif word_score == max_score:
+            # for tie breaking rules - prefer the word with the fewest letters
+            # unless one word has 10 letters, if top score is tied with 10 letters
+            # over the one with fewer tiles
+            #if there are multiple words that are the same score and the same length
+            #pick the first one in the supplied list
+                if len(word) < len(max_word) or len(word) == (len(max_word) == 10):
```

Comment:
In my opinion, I see that you're trying to solve two problems at once in this conditional:

1. finding the word with the highest score and shortest length, and
2. finding the word with a length of ten

To keep going on this, I'd encourage you to split up those two problems. If you cut down this conditional to `if len(word) < len(max_word)`, then we can see your code passes the tests around shortest word cleanly.

Then, I'd focus on solving for the ten-letter-tie-breaker problem after this `for` loop.

At the end, if your solution is longer than you'd like, you can refactor/combine from there :)